File: Daily Test/Java/Sort Tickets - Slot Number & Ticket Number 28.11.2025.java
import java.util.*;

public class Main {
    static class Pair {
        int x, y;
        Pair(int x, int y) { this.x = x; this.y = y; }
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        List<Pair> pairs = new ArrayList<>();
        for (int i = 0; i < n; i++) {
            int x = sc.nextInt();
            int y = sc.nextInt();
            pairs.add(new Pair(x, y));
        }
        pairs.sort((a, b) -> {
            if (a.y != b.y) return Integer.compare(a.y, b.y);
            return Integer.compare(a.x, b.x);
        });
        for (Pair p : pairs) {
            System.out.println(p.x + " " + p.y);
        }
    }
}

File: Daily Test/C++/Sort Tickets - Slot Number & Ticket Number 28.11.2025.cpp
#include <iostream>
#include <vector>
#include <algorithm>
using namespace std;

int main() {
    int n;
    cin >> n;
    vector<pair<int,int>> pairs(n);
    for (int i = 0; i < n; i++) {
        cin >> pairs[i].first >> pairs[i].second;
    }
    sort(pairs.begin(), pairs.end(),
         [](const pair<int,int>& a, const pair<int,int>& b) {
             if (a.second != b.second) return a.second < b.second;
             return a.first < b.first;
         });
    for (auto &p : pairs) {
        cout << p.first << " " << p.second << "
";
    }
    return 0;
}